src/install/install.zig

```diff
@@ -5589,16 +5658,17 @@ pub const PackageManager = struct {
         pub fn installEnqueuedPackages(
             this: *PackageInstaller,
             package_id: PackageID,
+            _: ExtractData,
             comptime log_level: Options.LogLevel,
         ) void {
             const name = this.lockfile.str(&this.names[package_id]);
             const resolution = this.resolutions[package_id];
-
-            if (this.manager.task_queue.fetchRemove(Task.Id.forNPMPackage(
-                Task.Tag.extract,
-                name,
-                resolution.value.npm.version,
-            ))) |removed| {
+            const task_id = switch (resolution.tag) {
+                .github => Task.Id.forTarball(this.manager.allocGitHubURL(resolution.value.github) catch unreachable),
```

Comment:
this leaks memory, we can avoid that by reusing a fixed size buffer

Reply:
Done in 7406e338db2471ead5ecf2ced8aaad4d80f47357